core/foundation/src/TClassEdit.cxx

```diff
@@ -535,16 +535,13 @@ ROOT::ESTLType TClassEdit::STLKind(std::string_view type)
       };
 
    // kind of stl container
-   auto len = type.length();
-   if (len) {
-      len -= offset;
-      for(int k=1;stls[k];k++) {
-         if (len == stllen[k]) {
-            if (type.compare(offset,len,stls[k])==0) return values[k];
-         }
+   const auto len = type.length() - offset;
```

Comment:
How do we survive `TClassEdit::STLKind("")`?

Reply:
same as before? `len` will be 0, we enter the loop but nothing matches, we end up returning `ROOT::kNotSTL`.

I can add an early return if you prefer

Reply:
> we enter the loop but nothing matches

Then why go through the loop?  I am indeed in favor of adding early return both before and after the offset calculation.

Reply:
ok -- why after?

Reply:
For "error case" like `"const "` and `"const std::"` .. probably mostly useless but cheap?

Reply:
alright!

Reply:
And I was just confused (but appreciate the early returns = clarification in code!)

Reply:
added a commit with the improvement

Reply:
...aaaand I screwed up, now fixing

Reply:
alright, done